packages/itwinui-react/src/core/Stepper/Stepper.tsx

```diff
@@ -69,22 +100,43 @@ export const Stepper = React.forwardRef((props, ref) => {
   return (
     <Box className={'iui-stepper'} ref={ref} {...rest}>
       <ol>
-        {steps.map((s, index) => (
-          <StepperStep
-            key={index}
-            index={index}
-            title={type === 'long' ? '' : s.name}
-            currentStepNumber={boundedCurrentStep}
-            totalSteps={steps.length}
-            type={type}
-            onClick={onStepClick}
-            description={s.description}
-          />
-        ))}
+        {steps.map((s, index) => {
+          const thisStepProps = stepProps?.(index);
```

Comment:
It should be possible to skip creating a variable and use props directly in component, right?
```
circleProps={circleProps?.(index)}
```

Reply:
I tried this and it was giving me overload errors. I'll keep it with the additional variable